cmd/osbuild-worker/jobimpl-osbuild.go

```diff
@@ -422,151 +412,29 @@ func (impl *OSBuildJobImpl) Run(job worker.Job) error {
-					log.Printf("koji logout failed: %v", err)
-				}
-			}()
-
-			if !osbuildOutput.Success {
-				err = k.CGFailBuild(int(options.BuildID), options.Token)
-				if err != nil {
-					log.Printf("CGFailBuild failed: %v", err)
-				}
-				continue
-			}
-
-			f, err := os.Open(path.Join(outputDirectory, options.Filename))
-			if err != nil {
-				r = append(r, err)
-				continue
-			}
-
-			hash, filesize, err := k.Upload(f, options.UploadDirectory, options.KojiFilename)
-			if err != nil {
-				r = append(r, err)
-				continue
-			}
-
-			hostOS, err := distro.GetRedHatRelease()
-			if err != nil {
-				r = append(r, err)
-				continue
-			}
-
-			build := koji.ImageBuild{
-				BuildID:   options.BuildID,
-				TaskID:    options.TaskID,
-				Name:      options.Name,
-				Version:   options.Version,
-				Release:   options.Release,
-				StartTime: start_time.Unix(),
-				EndTime:   end_time.Unix(),
-			}
-			buildRoots := []koji.BuildRoot{
-				{
-					ID: 1,
-					Host: koji.Host{
-						Os:   hostOS,
-						Arch: common.CurrentArch(),
-					},
-					ContentGenerator: koji.ContentGenerator{
-						Name:    "osbuild",
-						Version: "0", // TODO: put the correct version here
-					},
-					Container: koji.Container{
-						Type: "none",
-						Arch: common.CurrentArch(),
-					},
-					Tools: []koji.Tool{},
-					RPMs:  osbuildStagesToRPMs(osbuildOutput.Build.Stages),
-				},
-			}
-			output := []koji.Image{
-				{
-					BuildRootID:  1,
-					Filename:     options.KojiFilename,
-					FileSize:     uint64(filesize),
-					Arch:         common.CurrentArch(),
-					ChecksumType: "md5",
-					MD5:          hash,
-					Type:         "image",
-					RPMs:         osbuildStagesToRPMs(osbuildOutput.Stages),
-					Extra: koji.ImageExtra{
-						Info: koji.ImageExtraInfo{
-							Arch: "noarch",
-						},
-					},
-				},
-			}
-
-			_, err = k.CGImport(build, buildRoots, output, options.UploadDirectory, options.Token)
-			if err != nil {
-				r = append(r, err)
-				continue
-			}
+			osbuildJobResult.Success = true
+			osbuildJobResult.UploadStatus = "success"
 		default:
-			r = append(r, fmt.Errorf("invalid target type"))
+			err = fmt.Errorf("invalid target type")
+			appendTargetError(osbuildJobResult, err)
+			return nil
 		}
-	}
-
-	var targetErrors []string
-	for _, err := range r {
-		errStr := err.Error()
-		fmt.Printf("target errored: %s", errStr)
-		targetErrors = append(targetErrors, errStr)
-	}
-
-	var uploadstatus string = "failure"
-	if len(targetErrors) == 0 {
-		uploadstatus = "success"
-	}
-
-	err = job.Update(&worker.OSBuildJobResult{
-		Success:       osbuildOutput.Success && len(targetErrors) == 0,
-		OSBuildOutput: osbuildOutput,
-		TargetErrors:  targetErrors,
-		TargetResults: targetResults,
-		UploadStatus:  uploadstatus,
-	})
-	if err != nil {
-		return fmt.Errorf("Error reporting job result: %v", err)
+	} else if len(args.Targets) > 1 {
+		panic("The job request contains multiple upload targets. This is not supported!")
```

Comment:
If this occurs (if it's possible to occur), will the image be accessible after the panic/return?
If not, probably best to panic/return before running the build.

Reply:
It shouldn't be, but probably is :-D. So this makes sense! Thanks :-)